### DISCUSSION TITLE
v.0.5.0-beta

### DISCUSSION THREAD

**Discussion by awtkns** [reworkd/AgentGPT#587] [Announcements]
<h1 align="center">
🎉🤖&nbsp;&nbsp;&nbsp;Beta 0.5.0&nbsp;&nbsp;&nbsp;🤖🎉
</h1>

Welcome to the **Beta 5.0** release of AgentGPT! With this comes a barrel of features to go bananas over. Some of the highlights are:
- Migration to a [FastApi](https://github.com/tiangolo/fastapi) based backend which has replaced our NextJS backend. The maturity of Python LLM tooling was a key factor when making this decision.
- Addition of various tools such as Code Generation, Image Generation (via [Replicate](https://replicate.com/)) and Web Search
- Numerous styling and UI improvements

We will be releasing updates on a weekly schedule, stay tuned every Tuesday for the latest improvements to AgentGPT 🚀 If you'd like to help shape these updates join our [discord server](https://discord.gg/jdSBAnmdnY). We would love to hear your feedback. ❤️ 

##  👀 What's Changed
* ✨ Support for custom base URL by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/543
* ♻️ Refactor api by @awtkns in https://github.com/reworkd/AgentGPT/pull/555
* 💅 Update styling for chat window buttons by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/559
* 💅 Update ChatMessage styling by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/560
* ✨ Update analysis  by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/566
* 🛠️ Tools selector by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/569
* ✨ Sync by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/570
* 🌐 Enable web search by default by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/572
* 🐛 Fix "Conclude" task by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/574
* 🛠️ Select tools in the frontend by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/575
* 🐛 Fix mypy issues by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/576
* 💻 Add code writer tool by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/585

**Full Changelog**: https://github.com/reworkd/AgentGPT/compare/v.0.4.0-beta...v.0.5.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/reworkd/AgentGPT/releases/tag/v.0.5.0-beta'>v.0.5.0-beta</a>.</em>

**Comment by zhongqingsong** [comment]
Title is "🎉🤖   Beta 4.0   🤖🎉" , but now it's Beta 5.0. Once Again.

**Comment by yhyu13** [comment]
How is web search paid in 0.5? Do we still need to bring our own serp key?

**Reply by asim-shrestha** [reply]
Yeah exactly, not sure if there are any other ways to reliably do it. I know theres some self hostable search APIs

**Comment by yhyu13** [comment]
I found we are still summarizing web search results by serp's given snippets, right? I tried to add a crawler with puppeter but nodejs alwasy complain that native http request package is not supported. 

I think we should offload crawling to py backend

**Comment by yhyu13** [comment]
Another stupid question, how do we debug the py backend? I used VSCode auto attach debugger for npm client,
